core-java/src/main/java/eu/ostrzyciel/jelly/core/internal/DecoderLookup.java

```diff
@@ -0,0 +1,28 @@
+package eu.ostrzyciel.jelly.core.internal;
+
+import java.lang.reflect.Array;
+
+public class DecoderLookup<T> {
+
+    private int lastSetId = -1;
+    private final T[] lookup;
+
+    @SuppressWarnings("unchecked")
+    public DecoderLookup(Class<T> type, int maxEntries) {
+        this.lookup = (T[]) Array.newInstance(type, maxEntries);
```

Comment:
Ah, I see, you use the Class<T> here.

Look... maybe just cast the return values in get() to T? Sounds simpler and the JVM shouldn't mind. We can worry about the perf issues if they arise.

Reply:
Please add this to the "little book of performance angst to tackle later".